File: evolink/src/main/java/com/evolink/spring/model/CommunityPost.java
package com.evolink.spring.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "community_posts")
public class CommunityPost {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String tag;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    private String imagePath;

    @Column(nullable = false)
    private String authorName;

    private String authorImagePath;

    private LocalDateTime createdAt = LocalDateTime.now();

    // Getter/Setter
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getTag() { return tag; }
    public void setTag(String tag) { this.tag = tag; }
    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }
    public String getImagePath() { return imagePath; }
    public void setImagePath(String imagePath) { this.imagePath = imagePath; }
    public String getAuthorName() { return authorName; }
    public void setAuthorName(String authorName) { this.authorName = authorName; }
    public String getAuthorImagePath() { return authorImagePath; }
    public void setAuthorImagePath(String authorImagePath) { this.authorImagePath = authorImagePath; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
} 

File: evolink/lib/models/community_post.dart
class CommunityPost {
  final String tag;
  final String content;
  final String? imagePath;
  final String authorName;
  final String? authorImagePath;
  final String? id;
  final DateTime? createdAt;

  CommunityPost({
    required this.tag,
    required this.content,
    this.imagePath,
    required this.authorName,
    this.authorImagePath,
    this.id,
    this.createdAt,
  });

  // JSON에서 객체 생성
  factory CommunityPost.fromJson(Map<String, dynamic> json) {
    return CommunityPost(
      id: json['id'],
      tag: json['tag'],
      content: json['content'],
      imagePath: json['imagePath'],
      authorName: json['authorName'],
      authorImagePath: json['authorImagePath'],
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt']) 
          : null,
    );
  }

  // 객체를 JSON으로 변환
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'tag': tag,
      'content': content,
      'imagePath': imagePath,
      'authorName': authorName,
      'authorImagePath': authorImagePath,
      'createdAt': createdAt?.toIso8601String(),
    };
  }

  // 복사본 생성 (일부 필드만 변경)
  CommunityPost copyWith({
    String? tag,
    String? content,
    String? imagePath,
    String? authorName,
    String? authorImagePath,
    String? id,
    DateTime? createdAt,
  }) {
    return CommunityPost(
      tag: tag ?? this.tag,
      content: content ?? this.content,
      imagePath: imagePath ?? this.imagePath,
      authorName: authorName ?? this.authorName,
      authorImagePath: authorImagePath ?? this.authorImagePath,
      id: id ?? this.id,
      createdAt: createdAt ?? this.createdAt,
    );
  }
} 